api/controller.py

```diff
@@ -1591,35 +1593,58 @@ def _lane_details(self, languages, audiences):
 
         return languages, audiences
 
-    def contributor(self, contributor_name, languages, audiences):
+    def contributor(
+        self, contributor_name, languages, audiences,
+        feed_class=AcquisitionFeed
+    ):
         """Serve a feed of books written by a particular author"""
         library = flask.request.library
         if not contributor_name:
             return NO_SUCH_LANE.detailed(_("No contributor provided"))
 
+        # contributor_name is probably a display_name, but it could be a
+        # sort_name. Pass it in for both fields and
+        # ContributorData.lookup() will do its best to figure it out.
+        contributor = ContributorData.lookup(
+            self._db, sort_name=contributor_name, display_name=contributor_name
+        )
+        if not contributor:
+            return NO_SUCH_LANE.detailed(
+                _("Unknown contributor: %s") % contributor_name
+            )
+
+        search_engine = self.search_engine
+        if isinstance(search_engine, ProblemDetail):
+            return search_engine
+
         languages, audiences = self._lane_details(languages, audiences)
 
         lane = ContributorLane(
-            library, contributor_name, languages=languages, audiences=audiences
+            library, contributor, languages=languages, audiences=audiences
+        )
+        facets = load_facets_from_request(
+            worklist=lane, base_class=ContributorFacets
         )
-
-        annotator = self.manager.annotator(lane)
-        facets = load_facets_from_request(worklist=lane)
         if isinstance(facets, ProblemDetail):
             return facets
+
         pagination = load_pagination_from_request()
         if isinstance(pagination, ProblemDetail):
             return pagination
+
+        annotator = self.manager.annotator(lane, facets)
```

Comment:
Previously `facets` was not being passed in, which is definitely a problem, but I don't know what effects it would have.